addons/im_livechat/static/src/core/common/thread_model_patch.js

```diff
@@ -68,4 +68,17 @@ patch(Thread.prototype, {
             ? _t("This livechat conversation has ended")
             : null;
     },
+    /**
+     * @override
+     * @param {import("models").Persona} persona
+     */
+    getPersonaName(persona) {
+        if (this.channel_type === "livechat" && persona.user_livechat_username) {
+            return persona.user_livechat_username;
+        }
+        if (persona.is_public) {
+            return this.thread.anonymous_name;
```

Comment:
thread is `this` here (also shouldn't we still display the persona if there is no `anonymous_name`)?